### PR TITLE
token probs for non HF loaders

### PR DESCRIPTION
**Pull request by sALTaccount**
Right now, your console gets flooded with errors if you try to use the new Logits tab with non-HF model loaders. It doesn't seem to be (at least easily) possible to hijack the exllama samplers as they are written in C, so I was only able to implement for without the "use samplers" checkbox

I think the bigger part of this PR is that it adds proper error handling with messages telling the user what went wrong (etc they tried to use logits + sampler on a non HF loader), along with an error message telling the user they don't have a model loaded

### CODE REVIEW

**Comment by kalomaze** [comment]
I get the same issues with llama.cpp so it seems like both major loaders have the issue. Which begs the question of what loader was even used if both exllama and llama.cpp don't work...

**Comment by oobabooga** [comment]
Transformers and the _HF loaders are the main ones.

**Comment by oobabooga** [comment]
It should be possible to extend this to llama.cpp using the code in this block:

https://github.com/oobabooga/text-generation-webui/blob/main/modules/llamacpp_hf.py#L122

**Comment by sALTaccount** [comment]
@oobabooga llama.cpp_HF already works with the logits tab, its the normal llama.cpp that doesn't work. I think it should be possible to hijack the non llama.cpp sampler, but it looks decently painful so I'm leaving that out of this PR.

**Comment by sALTaccount** [comment]
i added non-sampler llamacpp support though, and added proper error handling for llamacpp

**Comment by oobabooga** [comment]
Thanks a lot @sALTaccount
